packages/generator/src/enum-type/enum.ts

```diff
@@ -0,0 +1,81 @@
+/* Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. */
+
+import {
+  ClassDeclarationStructure,
+  EnumDeclarationStructure,
+  PropertyDeclarationStructure,
+  StructureKind
+} from 'ts-morph';
+import {
+  getComplexTypeFieldDescription,
+  getComplexTypePropertyDescription
+} from '../typedoc';
+import { VdmComplexType, VdmEnumType, VdmProperty } from '../vdm-types';
+import {
+  getGenericParameters,
+  createPropertyFieldInitializer
+} from '../generator-utils';
+
+export function enumTypeClass(enumType: VdmEnumType): EnumDeclarationStructure {
+  return {
+    kind: StructureKind.Enum,
+    name: enumType.typeName,
+    isExported: true,
+    members: enumType.members.map(m => ({
+      name: m,
+      value: m
+    }))
+  };
+}
+
+export function fieldTypeClass(
+  complexType: VdmComplexType
+): ClassDeclarationStructure {
+  return {
+    kind: StructureKind.Class,
+    name: `${complexType.fieldType}<EntityT extends Entity>`,
+    extends: `ComplexTypeField<EntityT, ${complexType.typeName}>`,
```

Comment:
I guess this is also a copy paste error and it should use the enum-field?

Reply:
same